src/schemathesis/schemas.py

```diff
@@ -159,8 +159,18 @@ def is_reference(item: Dict[str, Any]) -> bool:
     return "$ref" in item
 
 
-def filter_parameters(parameters: ParametersList, place: str) -> ParametersList:
-    return [parameter for parameter in parameters if parameter["in"] == place]
+def convert_parameters(parameters: List[Dict[str, Any]], place: str) -> Dict[str, Any]:
+    new: Dict[str, Any] = {"properties": {}, "additionalProperties": False, "type": "object"}
+    for parameter in parameters:
+        if parameter["in"] == place:
+            if parameter.get("required", False):
+                new.setdefault("required", [])
+                new["required"].append(parameter["name"])
+            new["properties"][parameter["name"]] = {}
+            for key, value in parameter.items():
+                if key not in ("required", "name", "in"):
+                    new["properties"][parameter["name"]][key] = value
```

Comment:
The left part is a bit long and has some nesting that we can avoid:

```
def convert_property(data):
    new_property = {}
    for key, value in parameter.items():
        if key not in ("required", "name", "in"):
            new_property[key] = value
    return new_property
```

then:

```
new["properties"][parameter["name"]] = convert_property(parameter)
```